src/appleseed.shaders/src/appleseed/as_subsurface.osl

```diff
@@ -0,0 +1,370 @@
+    ]],
+
+    output closure color out_outColor = 0
+    [[
+        string as_maya_attribute_name = "outColor",
+        string as_maya_attribute_short_name = "oc",
+        string widget = "null"
+    ]],
+    output closure color out_outMatteOpacity = 0
+    [[
+        string as_maya_attribute_name = "outMatteOpacity",
+        string as_maya_attribute_short_name = "om",
+        string widget = "null",
+        int as_maya_attribute_hidden = 1,
+        int gafferNoduleLayoutVisible = 0
+    ]]
+)
+{
+    if (raytype("shadow") || raytype("transparency"))
+    {
+        return;
+    }
+
+    if (in_enable_matte)
+    {
+        out_outMatteOpacity = in_matte_opacity *
+            in_matte_opacity_color * holdout();
+
+        out_outColor += out_outMatteOpacity;
+    }
+
+    string sss_profile;
+
+    if (in_sss_profile == 0)
+    {
+        sss_profile = "better_dipole";
+    }
+    else if (in_sss_profile == 1)
+    {
+        sss_profile = "directional_dipole";
+    }
+    else if (in_sss_profile == 2)
+    {
+        sss_profile = "gaussian";
+    }
+    else if (in_sss_profile == 3)
+    {
+        sss_profile = "normalized_diffusion";
+    }
+    else if (in_sss_profile == 4)
+    {
+        sss_profile = "standard_dipole";
+    }
+    else
+    {
+        sss_profile = "randomwalk";
+    }
+
+    normal Nn = normalize(in_bump_normal);
+
+    if (in_sss_amount * max(in_color) > 0.0)
+    {
+        color albedo = clamp(in_sss_amount * in_color, 0.01, 0.99);
+
+        out_outColor += as_subsurface(
+            sss_profile,
+            Nn,
+            albedo,
+            in_sss_mfp_scale * in_sss_mfp,
+            in_ior,
+            "fresnel_weight", in_fresnel_weight,
+            "zero_scattering_weight", in_zero_scattering_weight);
+    }
+
+    if (in_specular_weight > 0.0)
+    {
+        vector tangent = Tn;
+
+        if (in_anisotropy_amount > 0.0)
+        {
+            if (in_anisotropy_mode == 0)
+            {
+                if (isconnected(in_anisotropy_map))
+                {
+                    vector vector_map = normalize(
+                        (vector) in_anisotropy_map * 2.0 - 1.0);
+
+                    tangent = normalize(
+                        vector_map[0] * Tn +
+                        vector_map[1] * Bn + 
+                        vector_map[2] * Nn);
+                }
+            }
+            else
+            {
+                tangent = in_anisotropy_direction;
+            }
+
+            if (in_anisotropy_angle > 0.0)
+            {
+                tangent = rotate(
+                    tangent,
+                    in_anisotropy_angle * M_2PI,
+                    point(0),
+                    point(Nn));
+            }
+        }
+
+        out_outColor += in_specular_weight * as_glossy(
+            "ggx",
+            Nn,
+            vector(0), // tangent
+            in_specular_roughness,
+            0.0, // specular spread for Student-T MDF
```

Comment:
Align comments vertically? Maybe more readable. Also, `Student's t-MDF` is probably more correct :)